src/transformers/trainer.py

```diff
@@ -161,18 +161,6 @@ class Trainer:
             :func:`~transformers.get_linear_schedule_with_warmup` controlled by :obj:`args`.
     """
 
-    model: PreTrainedModel
-    args: TrainingArguments
-    data_collator: DataCollator
-    train_dataset: Optional[Dataset]
-    eval_dataset: Optional[Dataset]
-    compute_metrics: Optional[Callable[[EvalPrediction], Dict]] = None
-    prediction_loss_only: bool
-    tb_writer: Optional["SummaryWriter"] = None
-    optimizers: Tuple[torch.optim.Optimizer, torch.optim.lr_scheduler.LambdaLR] = None
-    global_step: Optional[int] = None
-    epoch: Optional[float] = None
-
```

Comment:
Why are we removing that part in both trainers?

Reply:
Aren't those useful for IDE support/autocompletion?

Reply:
They are class variables since the `Trainer` is not a dataclass. Class variables should not be used when it's for attributes that change for each instance. I'm guessing Trainer was once a dataclass and has been changed since then, but those args weren't removed.

The `__init__` is left untouched (that's why you don't see it in the diff, see [here](https://github.com/huggingface/transformers/blob/848744aefa1044666e11bd33c512f549c6c03c9d/src/transformers/trainer.py#L164) for Trainer) so you still get the tab-completion.